remote_attestation/rust/src/attestation/unattested_session.rs

```diff
@@ -0,0 +1,146 @@
+//
+// Copyright 2021 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+use crate::proto::{AttestationIdentity, AttestationInfo, AttestationReport};
+use anyhow::{anyhow, Context};
+
+/// Represents an attestation process, where current machine remotely attests to a remote peer and
+/// sends attestation info to it.
+pub struct UnattestedSelf {
+    /// PEM encoded X.509 certificate that signs TEE firmware key.
+    tee_certificate: Vec<u8>,
+}
+
+impl UnattestedSelf {
+    pub fn new(tee_certificate: &[u8]) -> Self {
+        Self {
+            tee_certificate: tee_certificate.to_vec(),
+        }
+    }
+}
+
+/// Represents an attestation process, where current machine remotely attests a remote peer and
+/// verifies its attestation info.
+pub struct UnattestedPeer {
+    /// Expected value of the peer's TEE measurement.
+    expected_tee_measurement: Vec<u8>,
+}
+
+impl UnattestedPeer {
+    pub fn new(expected_tee_measurement: &[u8]) -> Self {
+        Self {
+            expected_tee_measurement: expected_tee_measurement.to_vec(),
+        }
+    }
+}
+
+/// Represents an attestation process, where current machine and a remote peer remotely attest each
+/// other.
+pub struct MutuallyUnattested {
+    /// PEM encoded X.509 certificate that signs TEE firmware key.
+    tee_certificate: Vec<u8>,
+    /// Expected value of the peer's TEE measurement.
+    expected_tee_measurement: Vec<u8>,
+}
+
+impl MutuallyUnattested {
+    pub fn new(tee_certificate: &[u8], expected_tee_measurement: &[u8]) -> Self {
+        Self {
+            tee_certificate: tee_certificate.to_vec(),
+            expected_tee_measurement: expected_tee_measurement.to_vec(),
+        }
+    }
+}
+
+pub trait UnattestedSession {
+    fn get_attestation_info(&self, public_key_hash: &[u8]) -> Option<AttestationInfo>;
```

Comment:
Does "get" mean "generate" here? If so, could you rename it? I think we used to use the terminology assertion generator / verifier, is this what this is supposed to be too? And is this method dual of the method below? If so, could you clarify why they use AttestationInfo and AttestationIdentity respectively?